flepimop/gempyor_pkg/src/gempyor/shared_cli.py

```diff
@@ -0,0 +1,186 @@
+        show_default=True,
+        help="write csv output?",
+    ),
+    click.option(
+        "--write-parquet/--no-write-parquet",
+        default=True,
+        show_default=True,
+        help="write parquet output?",
+    ),
+    click.option(
+        "-j",
+        "--jobs",
+        envvar="FLEPI_NJOBS",
+        type=click.IntRange(min=1),
+        default=multiprocessing.cpu_count(),
+        show_default=True,
+        help="the parallelization factor",
+    ),
+    click.option(
+        "-n",
+        "--nslots",
+        envvar="FLEPI_NUM_SLOTS",
+        type=click.IntRange(min=1),
+        help="override the # of simulation runs in the config file",
+    ),
+    click.option(
+        "--in-id",
+        "in_run_id",
+        envvar="FLEPI_RUN_INDEX",
+        type=str,
+        show_default=True,
+        help="Unique identifier for the run",
+    ),
+    click.option(
+        "--out-id",
+        "out_run_id",
+        envvar="FLEPI_RUN_INDEX",
+        type=str,
+        show_default=True,
+        help="Unique identifier for the run",
+    ),
+    click.option(
+        "--in-prefix",
+        "in_prefix",
+        envvar="FLEPI_PREFIX",
+        type=str,
+        default=None,
+        show_default=True,
+        help="unique identifier for the run",
+    ),
+    click.option(
+        "-i",
+        "--first_sim_index",
+        envvar="FIRST_SIM_INDEX",
+        type=click.IntRange(min=1),
+        default=1,
+        show_default=True,
+        help="The index of the first simulation",
+    ),
+    click.option(
+        "--stochastic/--non-stochastic",
+        "stoch_traj_flag",
+        envvar="FLEPI_STOCHASTIC_RUN",
+        default=False,
+        help="Run stochastic simulations?",
+    ),
+    click.option(
+        "-s",
+        "--seir_modifiers_scenarios",
+        envvar="FLEPI_SEIR_SCENARIO",
+        type=str,
+        default=[],
+        multiple=True,
+        help="override/select the transmission scenario(s) to run",
+    ),
+    click.option(
+        "-d",
+        "--outcome_modifiers_scenarios",
+        envvar="FLEPI_OUTCOME_SCENARIO",
+        type=str,
+        default=[],
+        multiple=True,
+        help="override/select the outcome scenario(s) to run",
+    ),
+    click.option(
+        "-c",
+        "--config",
+        "config_filepath",
+        envvar="CONFIG_PATH",
+        type=click.Path(exists=True),
+        required=False,  # deprecated = ["-c", "--config"],
+        # preferred = "CONFIG_FILES...",
+        help="Deprecated: configuration file for this simulation",
+    ),
+]
+""" List of `click` options that will be applied by `option_config_files` """
+
+
+def option_config_files(function: callable) -> callable:
+    """`click` Option decorator to apply handlers for all options"""
+    reduce(lambda f, option: option(f), config_file_options, function)
+    return function
+
+
+def parse_config_files(
+    config_files: list[str],
+    config_filepath: str,
```

Comment:
Not a `str`, is a `pathlib.Path`. At least that's what the corresponding click argument/option suggests, you could coerce to a string before providing to this function.